usr/src/lib/nsswitch/files/common/getnetgrent.c

```diff
@@ -0,0 +1,879 @@
+				if (limit == 0) {
+					syntax_err++;
+					break;
+				}
+				p = limit;
+				while (isspace(*p)) {
+					p++;
+				}
+				if (*p == terminators[0]) {
+					/*
+					 * Successfully parsed this name and
+					 *   the separator after it (comma or
+					 *   right paren); leave p ready for
+					 *   next parse.
+					 */
+					p++;
+					if (start == limit) {
+						/* Wildcard */
+						triple[i] = 0;
+					} else {
+						*limit = '\0';
+						triple[i] = start;
+					}
+				} else {
+					syntax_err++;
+					break;
+				}
+			}
+
+			if (syntax_err) {
+/*
+ * ===> log it;
+ * ===> try skipping past next ')';  failing that, abandon the line;
+ */
+				break;	/* Abandon this line */
+			} else if (!(*func)(triple, iter_args, &result)) {
+				/* Return result, good or bad */
+				done = 1;
+				break;
+			}
+		}
+		/* End of inner loop over val[] */
+		free(val);
+		val = NULL;
+	}
+	/* End of outer loop (!done && ngt_next(ngt) != 0) */
+
+	ngt_destroy(ngt);
+	free(ngt);
+
+	if (done) {
+		return (result);
+	} else if (nfound > 0) {
+		/* ==== ? Should only do this if all the top-level groups */
+		/*	  exist in YP?					  */
+		return (NSS_SUCCESS);
+	} else {
+		return (NSS_NOTFOUND);
+	}
+}
+
+
+/*
+ * Code for setnetgrent()
+ */
+
+/*
+ * Iterator function for setnetgrent():  copy triple, add to be->all_members
+ */
+static int
+save_triple(ccp trippp[NSS_NETGR_N], void *headp_arg,
+    nss_status_t *return_val)
+{
+	struct grouplist	**headp = headp_arg;
+	struct grouplist	*gl;
+	enum nss_netgr_argn	i;
+
+	if (__nss_files_netgr_debug != NULL)
+		__nss_files_netgr_debug(
+		    "save_tripple: h=%s u=%s d=%s",
+		    trippp[0] ? trippp[0] : "*",
+		    trippp[1] ? trippp[1] : "*",
+		    trippp[2] ? trippp[2] : "*");
+
+	if ((gl = (struct grouplist *)malloc(sizeof (*gl))) == 0) {
+		/* Out of memory */
+		*return_val = NSS_UNAVAIL;
+		return (0);
+	}
+	for (i = NSS_NETGR_MACHINE;  i < NSS_NETGR_N;  i++) {
+		if (trippp[i] == 0) {
+			/* Wildcard */
+			gl->triple[i] = 0;
+		} else if ((gl->triple[i] = strdup(trippp[i])) == 0) {
+			/* Out of memory.  Free any we've allocated */
+			enum nss_netgr_argn	j;
+
+			for (j = NSS_NETGR_MACHINE;  j < i;  j++) {
+				if (gl->triple[j] != 0) {
+					free(gl->triple[j]);
+				}
+			}
+			*return_val = NSS_UNAVAIL;
+			return (0);
+		}
+	}
+	gl->gl_nxt = *headp;
+	*headp = gl;
+	return (1);	/* Tell top_down() to keep iterating */
+}
+
+static nss_status_t
+netgr_set(struct files_backend	*be, void *a)
+{
+	struct nss_setnetgrent_args *args = (struct nss_setnetgrent_args *)a;
+	struct files_getnetgr_be	*get_be;
+	nss_status_t		res;
+
+	get_be = (struct files_getnetgr_be *)malloc(sizeof (*get_be));
+	if (get_be == 0) {
+		return (NSS_UNAVAIL);
+	}
+
+	get_be->all_members = 0;
+	res = top_down(be, &args->netgroup, 1, save_triple,
+	    &get_be->all_members);
+
+	if (res == NSS_SUCCESS) {
+		get_be->ops		= getnetgr_ops;
+		get_be->n_ops		= sizeof (getnetgr_ops) /
+		    sizeof (getnetgr_ops[0]);
```

Comment:
```suggestion
		get_be->n_ops		= ARRAY_SIZE(getnetgr_ops);
```

(and include `<sys/sysmacros.h>`)